pandas/core/arrays/arrow/accessors.py

```diff
@@ -0,0 +1,193 @@
+"""Accessors for arrow-backed data."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from pandas.compat import pa_version_under7p0
+
+if not pa_version_under7p0:
+    import pyarrow as pa
+    import pyarrow.compute as pc
+
+    from pandas.core.dtypes.dtypes import ArrowDtype
+
+if TYPE_CHECKING:
+    from pandas import (
+        DataFrame,
+        Series,
+    )
+
+
+class StructAccessor:
+    """
+    Accessor object for structured data properties of the Series values.
+
+    Parameters
+    ----------
+    data : Series
+        Series containing Arrow struct data.
+    """
+
+    _validation_msg = (
+        "Can only use the '.struct' accessor with 'struct[pyarrow]' dtype, not {dtype}."
+    )
+
+    def __init__(self, data=None) -> None:
+        self._parent = data
+        self._validate(data)
+
+    def _validate(self, data):
+        dtype = data.dtype
+        if not isinstance(dtype, ArrowDtype):
+            raise TypeError(self._validation_msg.format(dtype=dtype))
```

Comment:
```suggestion
            # Raise AttributeError so that reflection handles non-struct Series correctly.
            raise AttributeError(self._validation_msg.format(dtype=dtype))
```